engine/table/src/test/java/io/deephaven/engine/table/impl/SourcePartitionedTableTest.java

```diff
@@ -166,10 +171,60 @@ public void testAddAndRemoveLocations() {
                         FindExceptionCause.isOrCausedBy(errors.get(0), IllegalStateException.class).isPresent());
     }
 
+    /**
+     * This is a test for PR 4537, where SourceTable removes itself from the wrong refresh provider
+     */
+    public void testRemoveAndFail() {
```

Comment:
This is not annotated as a test...
```suggestion
    @Test
    public void testRemoveAndFail() {
```